ibis/tests/expr/test_rules.py

```diff
@@ -290,6 +294,42 @@ def test_invalid_column_or_scalar(validator, value, expected):
         validator(value)
 
 
+@pytest.mark.parametrize(
+    ('check_table', 'value', 'expected'),
+    [
+        (table, "int_col", table.int_col),
+        (table, table.int_col, table.int_col),
+    ],
+)
+def test_valid_column_from(check_table, value, expected):
+    class Test:
+        table = check_table
+
+    validator = rlz.column_from("table")
+    assert validator(value, this=Test()).equals(expected)
+
+
+@pytest.mark.parametrize(
+    ('check_table', 'validator', 'value', 'expected'),
+    [
+        (table, rlz.column_from("not_table"), "int_col", IbisTypeError),
+        (table, rlz.column_from("table"), "col_not_in_table", IbisTypeError),
+        (
+            table,
+            rlz.column_from("table"),
+            similar_table.int_col,
+            IbisTypeError,
+        ),
+    ],
+)
+def test_invalid_column_from(check_table, validator, value, expected):
+    class Test:
+        table = check_table
+
+    with pytest.raises(expected):
```

Comment:
You can remove the `expected` parameter and `IbisTypeError` instead, since the exception type isn't varying.

I realize you probably copied the pattern from above, but no reason to keep that around (someone can fix that in another PR).